csrc/dynamic_transform.cpp

```diff
@@ -727,6 +728,13 @@ bool DynamicTransformConcretizer::propagateFromProducerToConsumer(
 
   auto def = consumer->definition();
 
+  std::vector<std::unordered_map<IterDomain*, IterDomain*>> c2p_maps;
+  for (auto producer : ir_utils::filterByType<TensorView>(def->inputs())) {
+    PairwiseRootDomainMap root_map(producer, consumer);
+    c2p_maps.push_back(
+        root_map.mapConsumerToProducer(consumer->domain(), producer->domain()));
+  }
```

Comment:
Precomputing c2p maps so they don't need to be recomputed for each consumer ID/producer pair.